privacy-on-beam/pbeam/mean_test.go

```diff
@@ -785,3 +826,128 @@ func TestFindConvertToFloat64Fn(t *testing.T) {
+	// In this stage the algorithm will typically randomly choose either contribution for partition 0 or contribution to partition 1.
+	// However, since partitions are specified, partition 1 is dropped and the algorithm will choose partition 0.
+	// The sum of 2 means should be equal to 150/51 + 0/50 = 150/51 ≈ 2.94 in both cases (unlike 150/51 + 150/51 ≈ 5.88, if no cross-partition contribution bounding is done).
+	// The difference between these numbers ≈ 2.94 and the tolerance (see below) is ≈ 0.04, so the test should catch if there was no cross-partition contribution bounding.
+	exactCount := 51.0
+	exactMean := 150.0 / exactCount
+	result := []testFloat64Metric{
+		{0, exactMean},
+	}
+	p, s, col, want := ptest.CreateList2(triples, result)
+	col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
+
+	maxContributionsPerPartition := int64(1)
+	maxPartitionsContributed := int64(1)
+	epsilon := 60.0
+	delta := 0.01
+	lower := 0.0
+	upper := 150.0
+
+	// Specify partition 0, so id = 0 is guaranteed to contribute to partition 0.
+	partitionsCol := CreateList(s, []int{0})
+	// ε is split by 2 for noise and for partition selection, so we use 2*ε to get a Laplace noise with ε.
+	pcol := MakePrivate(s, col, NewPrivacySpec(2*epsilon, delta))
+	pcol = ParDo(s, tripleWithFloatValueToKV, pcol)
+	got := MeanPerKey(s, pcol, MeanParams{
+		MaxPartitionsContributed:     maxPartitionsContributed,
+		MaxContributionsPerPartition: maxContributionsPerPartition,
+		MinValue:                     lower,
+		MaxValue:                     upper,
+		NoiseKind:                    LaplaceNoise{},
+		specifiedPartitions:          partitionsCol})
+
+	means := beam.DropKey(s, got)
+	sumOverPartitions := stats.Sum(s, means)
+	got = beam.AddFixedKey(s, sumOverPartitions) // Adds a fixed key of 0.
+
+	want = beam.ParDo(s, float64MetricToKV, want)
+
+	// Tolerance for the partition with an extra contribution which is equal to 150.
+	tolerance1, err := laplaceToleranceForMean(25, lower, upper, maxContributionsPerPartition, maxPartitionsContributed, epsilon, -3675.0, 51.0, exactMean) // ≈0.00367
+	if err != nil {
+		t.Fatalf("laplaceToleranceForMean: got error %v", err)
+	}
+	// Tolerance for the partition without an extra contribution.
+	tolerance2, err := laplaceToleranceForMean(25, lower, upper, maxContributionsPerPartition, maxPartitionsContributed, epsilon, -3700.0, 50.0, 0.0) // ≈1.074
+	if err != nil {
+		t.Fatalf("laplaceToleranceForMean: got error %v", err)
+	}
+	if err := approxEqualsKVFloat64(s, got, want, tolerance1+tolerance2); err != nil {
+		t.Fatalf("TestMeanPerKeyWithPartitionsPerPartitionContributionBounding: %v", err)
+	}
+	if err := ptest.Run(p); err != nil {
+		t.Errorf("TestMeanPerKeyWithPartitionsPerPartitionContributionBounding: MeanPerKey(%v) = %v, want %v, error %v", col, got, want, err)
+	}
+}
+
+// Checks that MeanPerKey with specified partitions
+// returns a correct answer for int input values.
```

Comment:
nit: COuld you format this so that the first line is "full" before moving to the second line?